rust/lance-index/src/vector/hnsw.rs

```diff
@@ -213,13 +217,37 @@ struct SchemaIndexMetadata {
     metric_type: String,
 }
 
-#[derive(Debug, Serialize, Deserialize)]
+#[derive(Debug, Clone, Serialize, Deserialize)]
 pub struct HnswMetadata {
     entry_point: u32,
     level_offsets: Vec<usize>,
 }
 
 impl HNSW {
+    pub fn empty() -> Self {
+        Self {
+            levels: vec![],
+            metric_type: MetricType::L2,
+            entry_point: 0,
+            use_select_heuristic: true,
+        }
+    }
+
+    /// The number of nodes in the graph.
+    pub fn len(&self) -> usize {
+        self.levels[0].len()
+    }
+
+    /// Whether the graph is empty.
+    pub fn is_empty(&self) -> bool {
+        self.len() == 0
+    }
+
+    /// Metric type of the graph.
+    pub fn metric_type(&self) -> MetricType {
+        self.metric_type
+    }
```

Comment:
Any chance you can use `DistanceType`?  We are slowly trying to move away from `MetricType`.  They should be aliases:

```
/// For backwards compatibility.
pub type MetricType = DistanceType;
```

Reply:
here `MetricType` is alias of `DistanceType`

Reply:
here `MetricType` is alias of `DistanceType`